### PR TITLE
[*/datadog] Update to opentelemetry-mapping-go v0.11.0

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- [*/datadog] Bump to opentelemetry-mapping-go v0.11.0. This includes:
  - Refactor related to upcoming metric:
    - DataDog/opentelemetry-mapping-go/pull/230
    - DataDog/opentelemetry-mapping-go/pull/231
    - DataDog/opentelemetry-mapping-go/pull/229
  - Host metadata updates:
    - DataDog/opentelemetry-mapping-go/pull/184
    - DataDog/opentelemetry-mapping-go/pull/225
- Use `logs.Translator` for logs implementation
- Set `MeterProvider` to noop to prevent sending metrics for now until we agree on the design. Note that I reverted one of the changelog items added in #29785, since it's no longer true that we add this metric.

### CODE REVIEW

File: .chloggen/mx-psi_update-tov0.11.0.yaml [hunk 1]
```diff
@@ -7,15 +7,22 @@ change_type: enhancement
 component: datadogexporter
 
 # A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
-note: Add `datadog.otlp_translator.metrics.missing_source` counter, which counts the number of metrics that are missing a source (e.g. hostname).
```
Comment (dmitryax) [inline .chloggen/mx-psi_update-tov0.11.0.yaml:10]: Is it expected that the old entry be dropped?

Comment (mx-psi) [reply .chloggen/mx-psi_update-tov0.11.0.yaml:10]: Yes, it's intentional: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c88b4312804f5a42eed3615a9e70e6d65ce4f9d3/exporter/datadogexporter/factory.go#L100-L102 we need to discuss internally some details about the metric, so I am dropping that changelog note for now

**Review by bogdandrutu — APPROVED** [review]

File: exporter/datadogexporter/logs_exporter.go [hunk 2]
```diff
@@ -23,13 +24,14 @@ import (
 )
 
 // otelTag specifies a tag to be added to all logs sent from the Datadog exporter
-const otelTag = "otel_source:datadog_exporter"
+const otelTag = "datadog_exporter"
```
Comment (songy23) [inline exporter/datadogexporter/logs_exporter.go:27]: Maybe rename this to `otelSource`?

**Review by songy23 — APPROVED** [review]

File: exporter/datadogexporter/logs_exporter.go (hunk 2 above)
Comment (mx-psi) [reply exporter/datadogexporter/logs_exporter.go]: Makes sense, done

**Review by codeboten — APPROVED** [review]

**Pull request merged**